packages/client/src/object/Attachment.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Attachment {
+  rid: string;
+}
+
+export class Attachment implements Attachment {
```

Comment:
Given these are the same name, what do we get by separating them?

Reply:
Its perhaps useless? In my mind I was buying something about assignability across multiple versions of the osdk by using the Attachment type carefully and only using the Attachment class when we're doing implementation stuff.